python/adjoint/objective.py

```diff
@@ -76,28 +79,38 @@ def place_adjoint_source(self,dJ,dt):
             dV = 1/self.sim.resolution * 1/self.sim.resolution
         else:
             dV = 1/self.sim.resolution * 1/self.sim.resolution * 1/self.sim.resolution
-        da_dE = 0.5*(dV * self.cscale)
-        scale = da_dE * dJ * 1j * 2 * np.pi * self.frequencies / np.array([self.time_src.fourier_transform(f) for f in self.frequencies]) # final scale factor
+        
+        # an ugly way to calcuate the scaled dtft of the forward source
+        y = np.array([self.time_src.swigobj.current(t,dt) for t in np.arange(0,T,dt)]) # time domain signal
+        signal_dtft = np.matmul(np.exp(1j*2*np.pi*self.frequencies[:,np.newaxis]*np.arange(y.size)*dt), y) # dtft
```

Comment:
This is missing a factor of ∆t/sqrt(2π) — see [this line](https://github.com/NanoComp/meep/blob/5eeb01b2a27798daeda1ce0710d19f0cdbbbc261/src/dft.cpp#L181) and also the manual: https://meep.readthedocs.io/en/latest/Introduction/#transmittancereflectance-spectra